internal/image/selector.go

```diff
@@ -38,11 +38,14 @@ const (
 	SelectionStrategySemVer SelectionStrategy = "SemVer"
 )
 
-// Selector is an interface for selecting a single image from a container image
+// Selector is an interface for selecting images from a container image
 // repository.
 type Selector interface {
 	// Select selects a single image from a container image repository.
 	Select(context.Context) (*Image, error)
```

Comment:
I'm wondering if we still need this. It seems it's only ever called from integration tests. Could we discover with a limit of 1 in those tests instead?

Reply:
This is probably my biggest finding. If we don't need this and we kill off the implementations of it, we jettison about 150 lines or so.

Reply:
And if we _do_ kill off `Select()`, should we _then_ consider renaming `Discover()` to `Select()`?

It's a little awkward if the role of a _Selector_ is to _discover_.

We could get this to where a _Selector_ still _selects_ and the only difference from before this PR to after is that is selects more than one.

Reply:
I went with the option of killing off the original `Select()` implementation in favor of the `Discover()` one (which is now called `Select()`. As the reason I _originally_ did not make the change to the previous method directly was because I was still figuring out if the original implementation would still be required.